app/src/main/java/com/example/stromtracker/ui/raeume/raeumeBearbeiten_Loeschen/RaeumeBearbeitenLoeschenFragment.kt

```diff
@@ -70,51 +109,66 @@ class RaeumeBearbeitenLoeschenFragment(private var currRaum: Raum) : Fragment()
-                        raeumeViewModel.deleteRaeume(currRaum)
-                        //Man wir nur weitergeleitet, wenn man wirkllich löschen will. Deswegen nur bei positiv der Fragmentwechsel.
-                        //neues Fragment erstellen auf das weitergeleitet werden soll
-                        val frag = RaeumeFragment()
-                        //Fragment Manager aus Main Activity holen
-                        val fragMan = parentFragmentManager
-                        //Ftagment container aus content_main.xml muss ausgeählt werden, dann mit neuen Fragment ersetzen, dass oben erstellt wurde
-                        fragMan.beginTransaction().replace(R.id.nav_host_fragment, frag).commit();
-                        //und anschließend noch ein commit()
-                        dialog.cancel()
-                    })
+                    R.string.ja
+                ) { dialog, _ ->
+                    //Alle Geräte die dem aktuellen Raum hinzugefügt sind, werden dem Sonstigeraum zugeordnet
+                    raeumeViewModel.updateGeraeteByRaumId(currRaum.getRaumID(),sonstigesraumid)
+                    //Daten werden aus der Datenbank gelöscht
+                    //Daten aus Datenbank löschen
+                    raeumeViewModel.deleteRaeume(currRaum)
+                    //Man wir nur weitergeleitet, wenn man wirkllich löschen will. Deswegen nur bei positiv der Fragmentwechsel.
+                    //neues Fragment erstellen auf das weitergeleitet werden soll
+                    val frag = RaeumeFragment()
+                    //Fragment Manager aus Main Activity holen
+                    val fragMan = parentFragmentManager
+                    //Ftagment container aus content_main.xml muss ausgeählt werden, dann mit neuen Fragment ersetzen, dass oben erstellt wurde
+                    fragMan.beginTransaction().replace(R.id.nav_host_fragment, frag).commit()
+                    //und anschließend noch ein commit()
+                    dialog.cancel()
+                }
 
                 builder1.setNegativeButton(
-                    R.string.nein,
-                    DialogInterface.OnClickListener { dialog, id -> dialog.cancel() })
+                    R.string.nein
+                ) { dialog, _ -> dialog.cancel() }
 
                 val alert11: AlertDialog = builder1.create()
                 alert11.show()
-
-
             }
 
         }
-
         return root
     }
 
+    private fun customTextListener(edit: EditText): EditText {
+        edit.addTextChangedListener(object : TextWatcher {
+            override fun afterTextChanged(s: Editable) {
+                if (edit.text.toString() == "Sonstiges") {
+                    savebutton.visibility = View.INVISIBLE
+                    informationfield.text=raumnamesonstiges
+                }
+                else {
+                    savebutton.visibility = View.VISIBLE
+                    informationfield.text=""
+                }
```

Comment:
_Ist der Name "Sonstiges" dann eindeutig? Oder wie verhält sich das wenn man mehrere Räume mit Namen "Sonstiges" hat. Kann man die dann alle nicht mehr Editieren / Löschen? Oder wird es nicht akzeptiert einen Raum namens "Sonstiges" zu erstellen?_
Ich meinte damit nicht, dass du von .equals("...") zu == wechseln solltest.
Meine Frage ist: Sollte man es nicht blockieren, dass der Nutzer einen Raum mit dem Namen "Sonstiges" erstellt, da man diesen ja sonst nicht mehr bearbeiten oder löschen kann?

Reply:
Also das Wechseln ist empfohlen von Kotlin gewesen. Ich habe alle Vorschläge umgeändert und man kann davon ausgehen, dass es nur einen Raum Sonstiges pro Haushalt gibt, da man werder einen Raum Sonstiges erstellen kann, noch einen bestehenenden in diesen Raum umändern noch diesen Raum löschen kann. Deswegen ist das hier eindeutig.